packages/xo-server/src/xo-mixins/scheduling.mjs

```diff
@@ -92,21 +94,41 @@ export default class Scheduling {
     if (schedule === undefined) {
       throw noSuchObject(id, 'schedule')
     }
-    return schedule.properties
+    return {
+      ...schedule.properties,
+      healthCheckVmsWithTags:
+        schedule.properties.healthCheckVmsWithTags !== undefined
+          ? JSON.parse(schedule.properties.healthCheckVmsWithTags)
+          : undefined,
+    }
   }
 
   async getAllSchedules() {
-    return this._db.get()
+    return (await this._db.get()).map(schedule => ({
+      ...schedule,
+      healthCheckVmsWithTags:
+        schedule.healthCheckVmsWithTags !== undefined ? JSON.parse(schedule.healthCheckVmsWithTags) : undefined,
+    }))
   }
 
   async deleteSchedule(id) {
     this._stop(id)
     await this._db.remove(id)
   }
 
-  async updateSchedule({ cron, enabled, id, jobId, name, timezone, userId }) {
+  async updateSchedule({ cron, enabled, healthCheckSr, healthCheckVmsWithTags, id, jobId, name, timezone, userId }) {
     const schedule = await this.getSchedule(id)
-    patch(schedule, { cron, enabled, jobId, name, timezone, userId })
+    patch(schedule, {
+      cron,
+      enabled,
+      // null to delete the key of the object in case we remove healthcheck
+      healthCheckSr: healthCheckVmsWithTags !== undefined ? healthCheckSr : null,
+      healthCheckVmsWithTags: JSON.stringify(healthCheckVmsWithTags) ?? null,
```

Comment:
I believe that does not work.

Reply:
Why ?
[patch](https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server/src/patch.mjs#L6) remove the key of the object in case the value is `null`

Reply:
Ok, it works because `JSON.stringify(undefined) === undefined` which I find really weird.